src/viewer.js

```diff
@@ -373,9 +389,43 @@ $.Viewer = function( options ) {
         if (!_this._updateRequestId) {
             _this._updateRequestId = scheduleUpdate( _this, updateMulti );
         }
+
+        var tiledImage = event.item;
+        var fullyLoadedHandler =  function() {
+            var newFullyLoaded = _this._areAllFullyLoaded();
+            if (newFullyLoaded !== _this._fullyLoaded) {
+                _this._fullyLoaded = newFullyLoaded;
+                _this.raiseEvent('fully-loaded-change', { fullyLoaded: newFullyLoaded });
+            }
+        };
+        tiledImage._fullyLoadedHandler = fullyLoadedHandler;
```

Comment:
Let's make this property name even more specific (since it's kind of a weird case): `_fullyLoadedHandlerForViewer`.